### ISSUE TITLE
Research the security or risks around using personal devices as development/test environments.

### ISSUE THREAD

**Issue by steebe** [lshadler/indecision-plaza#14]
**Overview**
* There are inherent risks to testing new applications on personal devices. They should be addressed and a best practices document should be created that identifies how to circumvent or deal with such issues.

**Done When**
1. A document has been added to a resources directory within the project explaining the best way to deploy to a personal device for testing.

**Comment by steebe** [comment]
In regards to bare-metal vs. emulation testing a mobile application, there seem to be little to no _security_ risks involved with either approach on personal devices. Obviously, emulation seems safer up front, but on the topic of system calls to a mobile device's hardware resources, it appears that both platforms have the necessary protections within their API to prevent user-level code from making a call it shouldn't. 

This is not to say that there are not other inherent risks in testing a custom application on one's own device. Physical resource consumption, while not at risk of damage, is still at risk of throttling. Data usage is at risk of surpassing a "safe" amount and battery drain is a critical attribute to monitor. Each of these risks is something that every release should have tested on a physical device. 

These other noted concerns are brought up quite often in Mobile App testing literature, and it is because of them that there is an overwhelming amount of evidence that physical device testing is preferable over emulation when possible. The notion that functionality tests are handled by an em, as well as testing on a variant of "screen" sizes or older phone OS releases is handy and important, but not end-to-end.

In short, the benefits of testing on a personal mobile device far outweigh any threats to the device and information it holds. I recommend that every _milestone_ release should not reach a user until fully tested by the team on physical devices.

**Resources:**
[Emulation Risk](https://www.qualitestgroup.com/blog/test-methodologies-and-philosophies/thank-using-phone-emulation-risk/)
[Test Suggestions](https://arctouch.com/2015/08/9-top-mobile-app-testing-tips/)

**Comment by TheResults** [comment]
LGTM. After making a wiki page for security protocol containing this information, this issue should be free to close.
